clusterloader2/testing/load/configmap.yaml

```diff
@@ -6,7 +6,36 @@ metadata:
 immutable: true
 {{end}}
 data:
+{{if .EnableDNSTests}}
+  all-queries: |
+   kubernetes A
+   kubernetes AAAA
+   kubernetes.cluster.local A
+   kubernetes.cluster.local AAAA
+   kubernetes.svc.cluster.local A
+   kubernetes.svc.cluster.local AAAA
+   kubernetes.default.svc.cluster.local A
+   kubernetes.default.svc.cluster.local AAAA
+   google.com.svc.cluster.local A
+   google.com.svc.cluster.local AAAA
+   google.com.default.svc.cluster.local A
+   google.com.default.svc.cluster.local AAAA
+   google.com.cluster.local A
+   google.com.cluster.local AAAA
+   google.com A
+   google.com AAAA
+   1-2-3-4.default.pod.cluster.local A
+   metrics-server A
+   metrics-server AAAA
+   metrics-server.cluster.local A
+   metrics-server.cluster.local AAAA
+   metrics-server.svc.cluster.local A
+   metrics-server.svc.cluster.local AAAA
+   metrics-server.kube-system.svc.cluster.local A
+   metrics-server.kube-system.svc.cluster.local AAAA
+{{else}}
```

Comment:
I'm wondering if we should be "else-ing" here.
The current data was simply something artifical just to have something. If we have usecase for something else, which is clearly here, I'm with with simply changing this to your data.

Reply:
In fact I like the clear distinction between the case when we expect this configmap to contain some useful data and case when this can be a random stuff, as we have it with changing this conditionally.

I understand that removing this else will make this configmap easier to understand from coding perspective, but this hides dependency between deployment usage and configmap content. The current state is more complicated from coding perspective, but the dependency between configmap content and deployment is clear.

E.g if in future someone wants to implement a change where some percentile of deployments will need yet another configmap content, it will be much simpler to reason about when we can change the random data to something useful for that feature (it's clear that random data cannot be used anywhere).

So I recommend keeping the current state.

Reply:
I see what you're saying, but my counterargument for that is that, we shouldn't be changing the contents in that case, but instead:
- add new items to the map
- use those other items (mount just these other items at the pod template space) in this case

In other words, the configmap should be contatenation of everything that is potentially needed and we should solve what to mount at the pod template spec level.

Reply:
Wojtek volunteered to do that in https://github.com/kubernetes/perf-tests/pull/1546. Once it's merged, please rebase this PR and remove all "EnableDnsTests" in all configmap sections.

Reply:
That PR has just merged.

Reply:
Thanks, this simplifies the config quite a bit.